tcti/tcti_socket.c

```diff
@@ -528,3 +534,65 @@ TSS2_RC InitSocketTcti (
 
     return rval;
 }
+
+TSS2_RC Tss2_Tcti_Socket_Init (
+    TSS2_TCTI_CONTEXT *tctiContext,
+    size_t *size,
+    const char *conf
+    )
+{
+    TCTI_SOCKET_CONF sock_conf = { 0 };
+    TSS2_RC rc;
+    UriParserStateA state;
+    UriUriA uri;
+    const char *uri_str = conf != NULL ? conf : TCTI_SOCKET_DEFAULT;
+    size_t range;
+    /* maximum 5 digits in uint16_t + 1 for \0 */
+    char port[6] = { 0 };
+    char hostname[HOST_NAME_MAX + 1] = { 0 };
+
+    state.uri = &uri;
+    if (uriParseUriA (&state, uri_str) != URI_SUCCESS) {
+        rc = TSS2_TCTI_RC_BAD_VALUE;
+        goto out;
+    }
+
+    /* extract host & domain name / fqdn */
+    range = uri.hostText.afterLast - uri.hostText.first;
+    if (range > HOST_NAME_MAX) {
+        rc = TSS2_TCTI_RC_BAD_VALUE;
+        goto out;
+    }
+    strncpy (hostname, uri.hostText.first, range);
+    sock_conf.hostname = hostname;
+
+    /* extract port number */
+    range = uri.portText.afterLast - uri.portText.first;
+    if (range < 5 && range > 0) {
+        strncpy (port, uri.portText.first, range);
+        sscanf (port, "%" SCNu16, &sock_conf.port);
+    } else if (range == 0) {
+        sock_conf.port = DEFAULT_SIMULATOR_TPM_PORT;
+    } else { /* range > 5 */
+        rc = TSS2_TCTI_RC_BAD_VALUE;
+        goto out;
+    }
+    rc = InitSocketTcti (tctiContext, size, &sock_conf, 0);
+out:
+    uriFreeUriMembersA (&uri);
+    return rc;
+}
+
+/* public info structure */
+static TSS2_TCTI_INFO tss2_tcti_info __attribute__ ((unused)) = {
```

Comment:
I think this should not require `__attribute__ ((unused))`. See also tss2_tcti_device 's info struct.

Reply:
Right, it is used in 597

Reply:
Good catch. This was a left over artifact from v2. PR updated.